actions/update-go-mod-version/entrypoint

```diff
@@ -31,15 +31,7 @@ function main() {
   fi
 
   echo "Setting go version to: '${go_version}'"
-  sed -i "s/go .*/go ${go_version}/g" go.mod
-
-  echo "setting toolchain to: '${go_version}'"
-  if grep -qE 'toolchain go(.*)' < go.mod; then 
-    sed -i "s/toolchain go.*/toolchain go${go_version}/g" go.mod
-  else
-    # Write the toolchain directive two lines below the go version
-    sed -i "/go ${go_version}/r"<(printf "\ntoolchain go%s\n" "${go_version}") go.mod
-  fi
+  sed -i "s/^go .*/go ${go_version}/g" go.mod
```

Comment:
We need the `^` in the regex otherwise we hit any line that happens to have `go ` in it - which is a bunch of packages 😬